baseapp/helpers.go

```diff
@@ -3,19 +3,42 @@ package baseapp
 import (
 	tmproto "github.com/tendermint/tendermint/proto/tendermint/types"
 
+	"github.com/cosmos/cosmos-sdk/codec"
+	codectypes "github.com/cosmos/cosmos-sdk/codec/types"
 	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/x/auth/tx"
 )
 
+// txEncoder creates a proto TxEncoder for testing purposes.
+func txEncoder(interfaceRegistry codectypes.InterfaceRegistry) sdk.TxEncoder {
+	marshaler := codec.NewProtoCodec(interfaceRegistry)
+	txCfg := tx.NewTxConfig(marshaler, tx.DefaultSignModes)
+
+	return txCfg.TxEncoder()
+}
```

Comment:
the reason i'm creating a new proto codec here is because I want to avoid adding a `txEncoder` field in the baseapp struct that would only be used in tests